internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -529,7 +570,12 @@ func TestComputeControlPlaneVersion(t *testing.T) {
 			// Control plane is not upgrading implies that controlplane.spec.version is equal to controlplane.status.version.
 			// Control plane is not scaling implies that controlplane.spec.replicas is equal to controlplane.status.replicas,
 			// Controlplane.status.updatedReplicas and controlplane.status.readyReplicas.
-			name:            "should return cluster.spec.topology.version if the control plane is not upgrading and not scaling",
+			name: "should return cluster.spec.topology.version if the control plane is not upgrading and not scaling",
+			runtimeClient: runtimeclienttest.NewFakeRuntimeClientBuilder().
```

Comment:
I would also suggest to only use a response field here and build the runtimeClient below in t.Run. Just a lot less verbose